src/sdam/description/server.rs

```diff
@@ -79,7 +79,7 @@ pub(crate) struct ServerDescription {
 
 impl PartialEq for ServerDescription {
     fn eq(&self, other: &Self) -> bool {
-        if self.address == other.address && self.server_type == other.server_type {
+        if self.address != other.address || self.server_type != other.server_type {
```

Comment:
This is an unrelated bug that I found while debugging some test issues while this was in progress. Obviously, having the same address and server type does not mean that server descriptions are not equal.